### PR TITLE
Rebrand `ibis_omniscidb` to `ibis_heavyai`

### PR DESCRIPTION
**Pull request by tupui**
I followed the name which was set for the package here.

- Rebrand `ibis_omniscidb` to `ibis_heavyai`
- Update infrastructure
- Use `heavyai` instead of `pyomnisci`
- Use `heavydb` instead of `omnisci`

### CODE REVIEW

**Comment by tupui** [comment]
I don't have credentials to this repo so cannot run workflows.

cc @andrewseidl @pearu

**Comment by tupui** [comment]
For now I cannot make the CI green. Existing tests are not working (1 csv test and the rest is only when running the ibis test suite itself using this backend. Also the new version has some breaking changes) and this has nothing to do with this PR. I fixed a few things but there is still works to do for follow ups.

If it's fine, I will open some issues for follow up work.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -94,17 +85,8 @@ Before you submit a pull request, check that it meets these guidelines:
...
-    $ pytest tests.test_expr.py
-```
 ### Releasing
 
-To cut a new release, go 
-[GitHub Releases](https://github.com/Quansight/qadmin/releases/new) 
-Add the information for the new release and then click on 
-"Publish Release", it trigger a CI job that will cut a release at
-PyPI. When the package is released at PyPI, it will trigger a new
+To create a new release, publish a new tag. I will trigger the CI and release
```
Comment (pearu) [inline CONTRIBUTING.md:90]:
```suggestion
To create a new release, publish a new tag. It will trigger the CI and release
```

**Review by pearu — APPROVED** [review]
The rebranding changes look good to me.

The non-green CI state is worrying though. Can you run the tests locally?

**Comment by tupui** [comment]
Thanks @pearu 

> The non-green CI state is worrying though. Can you run the tests locally?

* Yes I can run them and all tests present in this repo are green, except the csv one (same as here).
* Pre-commit failure is because of mypy wanting to run in the test folder and here I am not sure what is wrong with the config.
* There are some tests from ibis directly. So we register the backend with ibis and run the tests on our backend. 11 tests are failing. I did not investigate yet.
* Pre-release tests are failing because we are using private functions from ibis. And these functions got deleted.

**Pull request merged**